aspnet/fundamentals/static-files.rst

```diff
@@ -47,132 +50,88 @@ Now, let's say that you have a project hierarchy where the static files you wish
 
     - test.png
 
-In order for the user to access test.png, you can configure the static files middleware as follows:
+For a request to access *test.png*, configure the static files middleware as follows:
 
-.. code-block:: c#
-  :emphasize-lines: 5-9
+.. literalinclude:: static-files/sample/Startup.cs
+  :language: c#
+  :lines: 38-39,53-62
+  :emphasize-lines: 7-12
+  :dedent: 8
 
-  public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
-  {
-    ...
-    // Add MyStaticFiles static files to the request pipeline.
-    app.UseStaticFiles(new StaticFileOptions()
-    {
-        FileProvider = new PhysicalFileProvider(@"D:\Source\WebApplication1\src\WebApplication1\MyStaticFiles"),
-        RequestPath = new PathString("/StaticFiles")
-    });
-    ...
+A request to ``http://<yourApp>/StaticFiles/test.png`` will serve the *test.png* file.
 
-At this point, if the user enters an address of ``http://<yourApp>/StaticFiles/test.png``, the ``test.png`` image will be served.
+Static file authorization
+-------------------------
+
+The static file module provides **no** authorization checks. Any files served by it, including those under *wwwroot* are publicly available. If you want to serve files based on authorization you should:
+
+- Store them outside of *wwwroot* and any directory accessible to the static file middleware
+- Serve them through a controller action, returning a :dn:class:`~Microsoft.AspNetCore.Mvc.FileResult` where authorization is applied
```

Comment:
Good idea. I've added the project and included a link at the top.

- `View or download sample code <https://github.com/aspnet/Docs/tree/master/aspnet/fundamentals/static-files/sample>`__